subsys/bluetooth/controller/ll_sw/coex_ticker.c

```diff
@@ -0,0 +1,258 @@
+				ticks_delay, /* one-shot delayed timeout */
+				0, /* periodic timeout  */
+				0, /* periodic remainder */
+				0, /* lazy, voluntary skips */
+				0,
+				callback, /* handler for executing radio abort or */
+					 /* flash work */
+				context, /* the context for the flash operation */
+				NULL, /* no op callback */
+				NULL);
+
+	return err;
+}
+
+
+#if defined(CONFIG_BT_CTLR_DEBUG_PINS) || \
+	defined(CONFIG_BT_CTLR_DEBUG_PINS_CPUAPP)
+#define DEBUG_PORT NRF_P0
+#define DEBUG_PIN_INHIBIT BIT(23)
+#define DEBUG_PIN_IRQ BIT(25)
+#define DEBUG_PIN_MASK    (DEBUG_PIN_IRQ | DEBUG_PIN_INHIBIT)
+#define DEBUG_INIT() \
+	do { \
+		DEBUG_PORT->DIRSET = DEBUG_PIN_MASK; \
+		DEBUG_PORT->OUTCLR = DEBUG_PIN_MASK; \
+	} while (0)
+
+#define DEBUG_INHIBIT(flag) \
+	do { \
+		if (flag) { \
+			DEBUG_PORT->OUTSET = DEBUG_PIN_INHIBIT; \
+		} else { \
+			DEBUG_PORT->OUTCLR = DEBUG_PIN_INHIBIT; \
+		} \
+	} while (0)
+
+
+#define DEBUG_IRQ(flag) \
+	do { \
+		if (flag) { \
+			DEBUG_PORT->OUTSET = DEBUG_PIN_IRQ; \
+		} else { \
+			DEBUG_PORT->OUTCLR = DEBUG_PIN_IRQ; \
+		} \
+	} while (0)
+#else
+#define DEBUG_INIT()
+#define DEBUG_INHIBIT(flag)
+#define DEBUG_IRQ(flag)
+#endif
+
+
+#include <hal/nrf_gpio.h>
+static void time_slot_callback_work(uint32_t ticks_at_expire,
+					uint32_t ticks_drift,
+					uint32_t remainder,
+					uint16_t lazy, uint8_t force,
+					void *context)
+{
+	uint8_t instance_index;
+	uint8_t ticker_id;
+	const struct device *dev = context;
+	const struct coex_ticker_config *config = dev->config;
+
+	__ASSERT(ll_radio_state_is_idle(),
+		 "Radio is on during flash operation.\n");
+
+	/* Read inhibit pin */
+	if (gpio_pin_get_dt(&config->inhibit_spec) == 1) {
+		DEBUG_INHIBIT(1);
+
+		if (!ll_radio_state_is_idle()) {
+			ll_radio_state_abort();
+		}
+		/* Schedule another check for inhibit pin and abort any events scheduled */
+		time_slot_delay(ticker_ticks_now_get(),
+				HAL_TICKER_US_TO_TICKS(COEX_RADIO_WORK_RESCHEDULE_DELAY_US),
+				time_slot_callback_work,
+				context);
+	} else {
+		LOG_DBG("COEX Inhibit Off");
+		DEBUG_IRQ(0);
+		DEBUG_INHIBIT(0);
+
+		/* Enable coex pin interrupt */
+		gpio_pin_interrupt_configure_dt(&config->inhibit_spec, GPIO_INT_EDGE_TO_ACTIVE);
+		ll_timeslice_ticker_id_get(&instance_index, &ticker_id);
+
+		/* Stop the time slot ticker */
+		_ticker_stop(instance_index, 0, ticker_id);
+	}
+}
+
+static void time_slot_callback_abort(uint32_t ticks_at_expire,
+					 uint32_t ticks_drift,
+					 uint32_t remainder,
+					 uint16_t lazy, uint8_t force,
+					 void *context)
+{
+	ll_radio_state_abort();
+	time_slot_delay(ticks_at_expire,
+			HAL_TICKER_US_TO_TICKS(COEX_RADIO_WORK_DELAY_US),
+			time_slot_callback_work,
+			context);
+}
+
+static int coex_ticker_inhibit_start(const struct device *dev)
+{
+	const struct coex_ticker_config *cfg = dev->config;
+	uint8_t instance_index;
+	uint8_t ticker_id;
+	uint32_t err;
+
+	ll_timeslice_ticker_id_get(&instance_index, &ticker_id);
```

Comment:
Have a new interface in controller and have a unique instance for co-ex, so that the ticker instance does not overlap flash driver use.